src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,39 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Parser {
+    Scanner in;
+    TaskList taskList;
+
+    Parser(TaskList taskList) {
+        in = new Scanner(System.in);
+        this.taskList = taskList;
+    }
+
+    void run() {
+        while (in.hasNextLine()) {
+            try {
+                String command = in.nextLine();
+                if (command.equals("bye")) {
+                    Ui.exit();
+                    break;
+                } else if (command.equals("list")) {
+                    taskList.list();
+                } else if (command.startsWith("done")) {
+                    taskList.markDone(command.substring(5));
```

Comment:
I think the line below has more than 120 characters. Perhaps it might good to split this line up into two and place half in a new line?